src/seraphis_main/contextual_enote_record_types.h

```diff
@@ -241,22 +265,24 @@ rct::xmr_amount amount_ref(const SpContextualEnoteRecordV1 &record);
 // ContextualBasicRecordVariant
 // - variant of all contextual basic enote record types
 //
-// origin_context_ref(): get the record's origin context
+// origin_status_ref(): get the record's origin status
+// transaction_id_ref(): get the record's transaction id
+// block_index_ref(): get the record's block index
 ///
 using ContextualBasicRecordVariant = tools::variant<LegacyContextualBasicEnoteRecordV1, SpContextualBasicEnoteRecordV1>;
-const SpEnoteOriginContextV1& origin_context_ref(const ContextualBasicRecordVariant &variant);
+std::uint64_t block_index_ref(const ContextualBasicRecordVariant &variant);
+const rct::key& transaction_id_ref(const ContextualBasicRecordVariant &variant);
+SpEnoteOriginStatus origin_status_ref(const ContextualBasicRecordVariant &variant);
```

Comment:
Nitpick: the comment for `origin_status_ref()` should be swapped with the comment for `block_index_ref()` to match the order of declarations below and definitions in .cpp .